api/report/undoc/b_api.c

```diff
@@ -856,24 +863,88 @@ int IodeEstimate(char* veqs, char* afrom, char* ato)
 // --------------------
 
 /**
- *  Simulate of model.
+ *  Decompose a model in its Connex Components. Each component is stored in a 
+ *  list of endo names. 
+ *  Optionnaly reorder the recursive component by applying nbtris times a heuristic method
+ *  that puts the endogenous variables as close as possible to the incidence matrix diagonal.
  *  
- *  @param [in] per_from      char*     per_from
- *  @param [in] per_to        char*     per_to
- *  @param [in] eqs_list      char*     eqs_list
- *  @param [in] endo_exo_list char*     endo_exo_list
- *  @param [in] eps           double    eps
- *  @param [in] relax         double    relax
- *  @param [in] maxit         int       maxit 
- *  @param [in] init_values   int       init_values
- *  @param [in] sort_algo     int       sort_algo
- *  @param [in] nb_passes     int       nb_passes
- *  @param [in] debug         int       debug
- *  @param [in] newton_eps    double    newton_eps
- *  @param [in] newton_maxit  int       newton_maxit
- *  @param [in] newton_debug  int       newton_debug      
- *  @return                   int       0 in success, -1 on error
+ *  @param [in] nbtris         int number of passes of the pseudo-triangulation method
+ *  @param [in] pre_listname   char*  name of the list to store the pre-recursive equation names
+ *  @param [in] inter_listname char*  name of the list to store the recursive equation names
+ *  @param [in] post_listname  char*  name of the list to store the post-recursive equation names
+ *  @param [in] eqs_list       char*  model equations names (opt.)
+ *  @return                    int    0 on success, -1 on error   
  */
+int IodeModelCalcSCC(int nbtris, char* pre_listname, char* inter_listname, char* post_listname, char *eqs_list)
+{
+    int     rc = -1;
+    KDB		*tdbe = NULL;
+    char    **eqs = NULL;
+
+    if(eqs_list && eqs_list[0] != 0) {
+        eqs = (char**) SCR_vtoms(eqs_list, ",; ");
+    }    
+
+    if(SCR_tbl_size(eqs) == 0)
+        tdbe = K_WS[K_EQS];
+    else
+        tdbe = K_quick_refer(K_WS[K_EQS], eqs);
+
+    rc = KE_ModelCalcSCC(tdbe, nbtris, pre_listname, inter_listname, post_listname);
+
+    if(SCR_tbl_size(eqs) != 0) K_free_kdb(tdbe);
+
+    SCR_free_tbl(eqs);
+    return(rc);
+}
+
+int IodeModelSimulateSCC(char *per_from, char *per_to, 
```

Comment:
Here, there are some differences:
- global parameters not passed as arguments: 
```
double eps, double relax, int maxit, 
int init_values, int debug, 
double newton_eps, int newton_maxit, int newton_debug)
```
- use of K_refer() instead of K_quick_refer()

Reply:
I've added a TODO in the header to add the above parameters as arguments.

Reply:
Could you write your comment above in the C++ code of Simulation please (as comments).